ts/features/bonus/bpd/components/bpdCardComponent/BpdCardComponent.tsx

```diff
@@ -111,46 +118,92 @@ type BadgeDefinition = {
 export const BpdCardComponent: React.FunctionComponent<Props> = (
   props: Props
 ) => {
-  const formatStatusBadge = (): BadgeDefinition => {
-    const { period } = props;
+  const { period, totalAmount } = props;
 
-    switch (period.status) {
-      case "Active":
-        return {
-          style: {
-            backgroundColor: IOColors.white
-          },
-          label: I18n.t("bonus.bpd.details.card.status.active")
-        };
-      case "Closed":
-        return {
-          style: {
-            backgroundColor: IOColors.black
-          },
-          label: I18n.t("bonus.bpd.details.card.status.closed")
-        };
-      case "Inactive":
-        return {
-          style: {
-            backgroundColor: IOColors.aqua
-          },
-          label: I18n.t("bonus.bpd.details.card.status.inactive")
-        };
-      default:
-        return {
-          style: {
-            backgroundColor: IOColors.white
-          },
-          label: "-"
-        };
-    }
+  const INITIAL_BADGE_DEFINITION = {
+    style: {
+      backgroundColor: IOColors.white
+    },
+    label: "-"
   };
+  const [amount, setAmount] = React.useState(["0", "00"]);
+  const [statusBadge, setStatusBadge] = React.useState<BadgeDefinition>(
+    INITIAL_BADGE_DEFINITION
+  );
+  const [showLock, setShowLock] = React.useState(false);
+  const [isInGracePeriod, setIsInGracePeriod] = React.useState(false);
 
-  const amount = formatNumberCentsToAmount(
-    props.totalAmount.totalCashback
-  ).split(",");
+  const statusClosedHandler = () => {
+    const actualDate = new Date();
+    const endDate = new Date(period.endDate.getTime());
+    endDate.setDate(endDate.getDate() + period.gracePeriod);
+    setStatusBadge({
+      style: {
+        backgroundColor: IOColors.black
+      },
+      label: I18n.t("bonus.bpd.detail.status.closed")
+    });
+    setShowLock(false);
+    setIsInGracePeriod(
+      actualDate.getTime() >= period.endDate.getTime() &&
+        actualDate.getTime() <= endDate.getTime()
+    );
+    // TODO: Add missing supercashback logic for amount
+    setAmount(
+      totalAmount.transactionNumber < period.minTransactionNumber
+        ? ["0", "00"]
+        : formatNumberCentsToAmount(props.totalAmount.totalCashback).split(",")
+    );
```

Comment:
When we are in `gracePeriod` the amount should be `props.totalAmount.totalCashback` even if `      totalAmount.transactionNumber < period.minTransactionNumber` (we are waiting for the missing transactions that could complete the minimum required).
We will ask confirmation to the design team just to be sure.

Reply:
@fabriziofff What I was thinking about the `gracePeriod`, working on the full version of the card, is: should we add the wip badge here too as we do in the preview version?

Reply:
@CrisTofani seems that the badge should be visible also in the full version of the card (missing in the excel state):

https://pagopa.invisionapp.com/console/IO---Cashback---Ciclo-di-vita-ckg868s4m00on01zz5g1z105w/ckg868vak00gv012o2fne9z7v/play